### PR TITLE
Invalid validation for "if token then"

### PR DESCRIPTION
**Pull request by giulianok**
Bug fix for `stripToken` which wasn't stripping the token if you pass an empty string

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  The changelog has not been updated. Please update the changelog.
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  The version was not updated. Please update the version.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: server/JWT.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ let JWT = {
       var parts = req.headers.authorization.split(' ')
       if (parts.length === 2 && /^Bearer$/i.test(parts[0])) { return parts[1] }
       throw new Error('Format is Authorization: Bearer [token]')
-    } else if (req.param('token')) {
+    } else if (req.body.hasOwnProperty('token')) {
```
Comment (daedalus28) [inline server/JWT.js]: The token can also come on the query - which is why we are deleting req.query.token as well.

File: server/stripToken.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ let JWT = require('./JWT')
 module.exports = async (req, res, next) => {
   try {
     let token = JWT.getToken(req)
-    await JWT.setTokenPayload(req, token)
+    if (token) await JWT.setTokenPayload(req, token)
```
Comment (daedalus28) [inline server/stripToken.js:6]: Make sure you handle the cases where token is passed along but is blank

**Review by daedalus28 — CHANGES_REQUESTED** [review]

File: server/stripToken.js (hunk 2 above)
Comment (giulianok) [reply server/stripToken.js:6]: What do you mean?

Comment (daedalus28) [reply server/stripToken.js:6]: get token could return `''`

File: server/JWT.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ let JWT = {
       var parts = req.headers.authorization.split(' ')
       if (parts.length === 2 && /^Bearer$/i.test(parts[0])) { return parts[1] }
       throw new Error('Format is Authorization: Bearer [token]')
-    } else if (req.param('token')) {
+    } else if (req.query.hasOwnProperty('token') || req.body.hasOwnProperty('token')) {
```
Comment (daedalus28) [inline server/JWT.js]: how is this different from `req.param` which checks body and query?

Comment (giulianok) [reply server/JWT.js]: `req.param` can return null, undefined, empty string, etc and the IF statement won't pass

File: server/stripToken.js (hunk 2 above)
Comment (giulianok) [reply server/stripToken.js:6]: is that affect on something? if I don't have a token or an empty string I shouldn't run `setTokenPayload`

File: server/JWT.js (hunk 3 above)
Comment (daedalus28) [reply server/JWT.js]: got it, use req.allParams and a _.has

**Review by daedalus28 — APPROVED** [review]

**Pull request merged**